packages/neuron-wallet/src/services/tx/transaction-persistor.ts

```diff
@@ -297,22 +300,32 @@ export class TransactionPersistor {
       }
       return output
     })
-
-    const sliceSize = 100
+    let currentWalletInputs: InputEntity[] = inputs
+    let currentWalletOutputs: OutputEntity[] = outputs
+    let txLocks: TxLockEntity[] = []
+    if (lockArgsSet?.size) {
+      currentWalletInputs = inputs.filter(v => this.isCurrentWalletCell(v, lockArgsSet))
+      currentWalletOutputs = outputs.filter(v => this.isCurrentWalletCell(v, lockArgsSet))
+      txLocks = [...new Set(
+        [
+          ...inputs.filter(v => v.lockHash && !this.isCurrentWalletCell(v, lockArgsSet))
+                  .map(v => v.lockHash!),
+          ...outputs.filter(v => !this.isCurrentWalletCell(v, lockArgsSet))
+                  .map(v => v.lockHash),
```

Comment:
Perhaps using `!currentWalletInputs.includes(v)` to replace `!this.isCurrentWalletCell(v, lockArgsSet)` might be more readable and easier to maintain.
Just my personal thoughts, it's optional.

Reply:
I can't get the point. Where is the `currentWalletInputs ` from?

Reply:
It is in the lines of code not shown above:
```currentWalletInputs = inputs.filter(v => this.isCurrentWalletCell(v, lockArgsSet))```

Reply:
`currentWalletInputs` is the filter result, add a filter `!currentWalletInputs.includes(v)` to filter which?

Reply:
like this:
```
txLocks = [...new Set(
  [
    ...inputs.filter(v => v.lockHash && !currentWalletInputs.includes(v))
            .map(v => v.lockHash!),
    ...outputs.filter(v => !currentWalletOutputs.includes(v))
            .map(v => v.lockHash),
  ]
```

Reply:
In my opinion, `...inputs.filter(v => v.lockHash && !this.isCurrentWalletCell(v, lockArgsSet))` is more clearly and fast.
And it does not need to depend on `currentWalletInputs`

Reply:
Ok no problem, it just seems to me that it might improve readability